.pre_commit/generate-facets/generate.py

```diff
@@ -0,0 +1,253 @@
+# Copyright 2018-2024 contributors to the OpenLineage project
+# SPDX-License-Identifier: Apache-2.0
+from __future__ import annotations
+
+import copy
+import itertools
+import json
+import os
+import pathlib
+import re
+import shutil
+import tempfile
+from collections import defaultdict
+
+import click
+from datamodel_code_generator import DataModelType, PythonVersion
+from datamodel_code_generator.imports import Import
+from datamodel_code_generator.model import get_data_model_types
+from datamodel_code_generator.parser.jsonschema import JsonSchemaParser
+from datamodel_code_generator.types import Types
+
+_UNDER_SCORE_1 = re.compile(r"([^_])([A-Z][a-z]+)")
+_UNDER_SCORE_2 = re.compile("([a-z0-9])([A-Z])")
+
+
+def camel_to_snake(string: str) -> str:
+    subbed = _UNDER_SCORE_1.sub(r"\1_\2", string)
+    return _UNDER_SCORE_2.sub(r"\1_\2", subbed).lower()
+
+
+# import attrs instead of dataclass
+data_model_types = get_data_model_types(
+    DataModelType.DataclassesDataclass, target_python_version=PythonVersion.PY_38
+)
+new_model = data_model_types.data_model
+new_model.DEFAULT_IMPORTS = (
+    Import.from_full_path("attr.define"),
+    Import.from_full_path("attr.field"),
+)
+
+# locations definitions
+base_spec_location = pathlib.Path(__file__).resolve().parent.parent.parent / "spec" / "OpenLineage.json"
+facets_spec_location = pathlib.Path(__file__).resolve().parent.parent.parent / "spec" / "facets"
+default_output_location = (
+    pathlib.Path(__file__).resolve().parent.parent.parent
+    / "client"
+    / "python"
+    / "openlineage"
+    / "client"
+    / "generated"
+)
+templates_location = pathlib.Path(__file__).resolve().parent / "templates"
+python_client_location = pathlib.Path(__file__).resolve().parent.parent.parent / "client" / "python"
+
+# custom code
+set_producer_code = """
```

Comment:
Any reason why all of those variables are lowercase ?

Reply:
Same as above. Will change them to uppercase.